rust/main/agents/relayer/src/relayer.rs

```diff
@@ -1408,4 +1508,182 @@ mod test {
             .unwrap();
         assert_eq!(metric.get(), 1);
     }
+
+    async fn build_relayer(settings: RelayerSettings) -> eyre::Result<Relayer> {
+        let agent_metadata = AgentMetadata::new("relayer_git_hash".into());
+
+        let metrics = settings.as_ref().metrics("relayer")?;
+        let task_monitor = tokio_metrics::TaskMonitor::new();
+        let agent_metrics = AgentMetrics::new(&metrics)?;
+        let chain_metrics = ChainMetrics::new(&metrics)?;
+        let runtime_metrics = RuntimeMetrics::new(&metrics, task_monitor)?;
+
+        let (_, tokio_server) = console_subscriber::ConsoleLayer::new();
+
+        Relayer::from_settings(
+            agent_metadata,
+            settings,
+            metrics,
+            agent_metrics,
+            chain_metrics,
+            runtime_metrics,
+            tokio_server,
+        )
+        .await
+    }
+
+    /// Run relayer for 50s to ensure it doesn't crash
```

Comment:
I don't love that unit tests now take at least 50s, is this necessary?

Reply:
I guess I can run it, then keep checking for metrics. If metrics are up, then exit. If metrics aren't up after 50s, fail